pkg/apis/serving/v1alpha1/node_group_types.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2023 The KServe Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ModelCacheNodeGroupSpec defines a group of nodes for to download the model to.
+// +k8s:openapi-gen=true
+type ModelCacheNodeGroupSpec struct {
+	StorageLimit resource.Quantity `json:"storageLimit"`
+	NodeSelector map[string]string `json:"nodeSelector"`
```

Comment:
Discussed with @sukumargaonkar. Two approaches here.
Approach 1 - It's up to the user to make sure they use the correct label selector so two node groups never have overlaps.
Approach 2 - Require a specific label set on nodes. e.g. kserve-nodegroup-name: xyz
Otherwise if we allow overlapping in nodegroups. storage size calculation would be tricky.